orttraining/orttraining/python/training/ortmodule/ort_triton/kernel/_mm.py

```diff
@@ -0,0 +1,486 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import math
+import os
+from types import ModuleType
+from typing import Tuple
+
+import torch
+
+from onnxruntime.training import ortmodule
+
+from .._cache import ModuleCache, PyCodeCache
+from .._utils import next_power_of_2
+
+_DEBUG_MODE = ortmodule._defined_from_envvar("ORTMODULE_TRITON_DEBUG", 0) != 0
+
+
+_MM_TEMPLATE = """
+import torch
+import triton
+import triton.language as tl
+
+
+@triton.autotune(
+    configs=[
+{autotune_configs}
+    ],
+    key=["M", "N", "K"],
+)
+@triton.jit
+def {kernel_name}(
+    A, B, OUT, M, N, K, BLOCK_M: tl.constexpr, BLOCK_N: tl.constexpr, BLOCK_K: tl.constexpr, GROUP_M: tl.constexpr
+):
+    M = {M}
```

Comment:
Since it's generated code, we will try to use as more constant value as possible to make it faster. But we still need the M in the argument list because the Triton's autotune need it as a part of the key, which much be something from argument list.